lib/faktory/configuration.ex

```diff
@@ -3,28 +3,44 @@ defmodule Faktory.Configuration do
 
   def call(module, defaults) do
     config = Application.get_env(module.otp_app, module, [])
+
     if config[:configured] do
       Keyword.delete(config, :configured)
     else
-      config = defaults
-      |> Keyword.merge(config)
-      |> module.init
-      |> put_wid(module.type) # Client connection don't have wid.
-      |> resolve_all_env_vars
-      |> Keyword.put_new(:module, module)
-      |> Keyword.put(:jobtype_map, jobtype_map(module.otp_app))
-      |> Keyword.merge(Faktory.get_env(:cli_options) || [])
-      |> normalize
-      |> Keyword.put(:configured, true)
+      config =
+        defaults
+        |> Keyword.merge(config)
+        |> module.init
+        # Client connection don't have wid.
+        |> put_wid(module.type)
+        |> resolve_all_env_vars
+        |> Keyword.put_new(:module, module)
+        |> Keyword.put(:jobtype_map, jobtype_map(module.otp_app))
+        |> Keyword.merge(Faktory.get_env(:cli_options) || [])
+        |> normalize
+        |> Keyword.put(:configured, true)
+
       Application.put_env(module.otp_app, module, config)
       call(module, defaults)
     end
   end
 
   defp jobtype_map(otp_app) do
     {:ok, modules} = :application.get_key(otp_app, :modules)
+
     Enum.reduce(modules, %{}, fn module, acc ->
-      behaviours = module.__info__(:attributes)[:behaviour] || []
+      behaviours =
+        cond do
+          Kernel.function_exported?(module, :__info__, 1) ->
+            module.__info__(:attributes)[:behaviour] || []
+
+          Kernel.function_exported?(module, :module_info, 1) ->
+            module.module_info(:attributes)[:behaviour] || []
```

Comment:
I ported this code to the branch I'm working on and it broke the tests. It seems that trying to call a function on a module will implicitly load the module, but `function_exported?/3` does not.

Here's how I worked around the issue:
https://github.com/cjbottaro/faktory_worker_ex/pull/58/files#diff-71ab03c9606179098bacbf7eeddbdba6R40-R43

Also, I'm thinking my branch will become 1.0.0 and all the PRs you're making should be merged in and released as 0.8.0.

Reply:
Great, thanks! Do you want me to apply these changes to the PR or are you happy continuing w/ your 0.8/1.0.0 as is?

Reply:
On the PR is good. I'm going to merge your PRs and release as 0.8. Then the branch I'm currently working on will become 1.0 since it has more breaking changes. Thanks!